python/cudf/cudf/core/index.py

```diff
@@ -231,7 +231,7 @@ def step(self):
     def _num_rows(self):
         return len(self)
 
-    @cached_property
+    @cached_property  # type: ignore
```

Comment:
Can we give this a return type like `def … () -> Union[…]:`?

Reply:
That doesn't help:

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

python/cudf/cudf/core/index.py:234: error: Decorated property not supported
```

Reply:
Okay, here is a second question. Can the `# type: ignore` comment be on the same line as `def` and not the property decorator?

Reply:
No. It needs to be on the property decorator line. It's also a terrible warning, since mypy can actually deduce the type (this appears like it was fixed recently) https://github.com/python/mypy/issues/1362